conans/model/conf.py

```diff
@@ -493,13 +490,28 @@ def set_relative_base_folder(self, folder):
 
     @staticmethod
     def _check_conf_name(conf):
-        if USER_CONF_PATTERN.match(conf) is None and conf not in BUILT_IN_CONFS:
-            raise ConanException(f"[conf] '{conf}' does not exist in configuration list. "
-                                 f" Run 'conan config list' to see all the available confs.")
+        if (CORE_CONF_PATTERN.match(conf) is not None or TOOLS_CONF_PATTERN.match(conf) is not None):
```

Comment:
It's a bit longer, but I think it's a lot clearer how it's checking it. Maybe duplication can be reduced if someone feels like it's needed, happy to do that